File: src/dotnet/content/JfYu.WebApi.Template/Entity/AuditLog.cs
using JfYu.Data.Model;
using System.ComponentModel.DataAnnotations;

namespace JfYu.WebApi.Template.Entity
{
    public class AuditLog : BaseEntity
    {
        public int? UserId { get; set; }

        [MaxLength(50)]
        public string? UserName { get; set; }

        [Required]
        [MaxLength(20)]
        public string Action { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string Resource { get; set; } = string.Empty;

        [MaxLength(50)]
        public string? ResourceId { get; set; }

        public string? OldValue { get; set; }

        public string? NewValue { get; set; }

        [MaxLength(50)]
        public string? IP { get; set; }
    }
}

File: src/vue/apps/web-antd/src/api/system/auditLog.ts
import { requestClient } from '#/api/request';

export namespace SystemAuditLogApi {
  export interface AuditLog {
    id: number;
    userId: null | number;
    userName: null | string;
    action: string;
    resource: string;
    resourceId: null | string;
    oldValue: null | string;
    newValue: null | string;
    ip: null | string;
    createdTime: string;
  }
}

export async function getAuditLogList(params: Record<string, any>) {
  return requestClient.get<{
    items: SystemAuditLogApi.AuditLog[];
    total: number;
  }>('/AuditLog', { params });
}
